File: maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int sum = 0;
        for (int j = 0; j < k; j++) {
            sum += nums[j];
        }
        int max = sum;

        for (int i = 1; i < nums.length - k + 1; i++) {
            sum = sum - nums[i - 1] + nums[i + k - 1];
            max = Math.max(max, sum);
        }
        return Double.valueOf(max) / k;
    }
}

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        presum = 0
        result = -float('inf')
        for i in range(0, k - 1):
            presum += nums[i]
        for i in range(k - 1, len(nums)):
            drop = 0 if i == k - 1 else nums[i - k]
            sum = presum + nums[i] - drop
            result = max(result, sum)
            presum = sum
        return result / k